internal/wasm/httpserver.go

```diff
@@ -0,0 +1,228 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package wasm
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/envoyproxy/gateway/internal/logging"
+)
+
+const (
+	serverHost        = "envoy-gateway"
+	serverPort        = 18002
+	maxFailedAttempts = 10
+	// TODO: zhaohuabing make this configurable
+	attemptsResetInterval = 5 * time.Minute
+	attemptResetDelay     = 1 * time.Hour
+)
+
+var _ Cache = &HTTPServer{}
+
+// HTTPServer wraps a local file cache and serves the Wasm modules over HTTP.
+type HTTPServer struct {
```

Comment:
can we model this as a runner ?
gateway-api can publish the IR, and this runner could subscribe to this layer